### PR TITLE
docs(config options): rewrite `transitiveRemediation`

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Rewrite `transitiveRemediation` config option docs:
  - List the only supported configuration conditions
  - _Emphasize_ the key points
  - Incorporate comments from `@rarkins` with slight rewrites (see #28451)

## Context

A user told us the `transitiveRemediation` docs are unclear in:

- #28451

This still needs a bit of a rewrite from the maintainers, to decide what's important and what not. I don't know enough to know what to drop/keep. I also want a double check on technical accuracy. :wink: 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
this feature will be dropped on v38, so we should add a warning instead ⚠️

**Comment by HonkingGoose** [comment]
Oh good catch! :heart:

This PR (merged into the `v38` branch) drops the `transitiveRemediation` feature:

- #27985

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -3877,12 +3877,30 @@ Please see the above link for valid timezone names.
...
+If you have _unpinned_ dependencies then you may get a _lockfile-only_ CVE remediation.
+For example:
+
+- Package file contains `^1.0.0`
+- Lock file contains `1.0.1`
+- CVE is fixed in `1.0.2`
+
+Then you will get a PR that updates the _lockfile_ from `1.0.1` to `1.0.2`.
+
+`transitiveRemediation` is considered a "feature flag": we aim to remove it as a config option and default to this behavior after it has been widely tested.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:3903]:
```suggestion
<!-- prettier-ignore -->
!!! warning
    We will remove the `transitiveRemediation` config option in the upcoming `v38` release of Renovate.
    We recommend that you stop using this config option.
```

**Comment by rarkins** [comment]
Sorry, transitiveRemediation is being removed in v38

**Pull request closed without merging**